torch/csrc/utils/python_dispatch.cpp

```diff
@@ -226,6 +226,13 @@ void initDispatchBindings(PyObject* module) {
     return states;
   });
 
+  m.def("_dispatch_tls_set_dispatch_key_excluded", [](const char* dispatch_key, bool desired_state) {
```

Comment:
I would argue that binding `ExcludeDispatchKeyGuard` will make the API harder to use wrong?
Or do we have a concern with making RAII in python due to potential cycle/gc?

Reply:
It's just more annoying to bind C++ RAII stuff. Actually as you can see from the usage later this isn't good anyway because we have a bunch of constants that are used in conjunction with the guards to actually grab the right dispatch keys. Some API design will be necessary.

Reply:
Well binding an RAII is pretty simple: https://github.com/pytorch/pytorch/blob/83d40a4dba54541340462301fc86118167608cb2/torch/csrc/autograd/init.cpp#L397-L398

It is used at https://github.com/pytorch/pytorch/blob/83d40a4dba54541340462301fc86118167608cb2/torch/overrides.py#L1935-L1940